packages/dashboard/src/components/iot-dashboard/iot-dashboard-internal.tsx

```diff
@@ -419,6 +558,21 @@ export class IotDashboardInternal {
             }}
           ></div>
         )}
+
+        {this.contextMenuOpen && this.contextMenuPosition && (
+          <iot-context-menu {...this.contextMenuPosition}>
+            {this.getContextMenuConfiguration().map((section) => (
```

Comment:
is there someway we could encapsulate all of this, including the construction of the configuration object into it's own component? i.e. `iot-dashboard-context-menu`

Reply:
Ya, I was going back and forth on creating a component to house all that. I think you're right, will make it more maintainable.